
## EDA(Exploratory Data Analysis)

#### a) Ler o dataset 

In [38]:
import pandas as pd

data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')
data1.to_csv('data1.csv', index = True)
print(data1.head())



print("-------------------------------------------info")

n_rows, n_col = data1.shape
print("Number of rows: ", n_rows)
print("Number of columns: ", n_col)

C:\Users\Renan\AppData\Local\Temp\ipykernel_1256\584370875.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')


                    id      date   hour   ddi country country_iso3 ddd  \
0   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
1   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
2  5788106393468158140  07/04/20  04:07  55.0  BRASIL          BRA  21   
3   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
4  5788106393468158140  07/04/20  04:13  55.0  BRASIL          BRA  21   

            state   group  midia  url                      text  
0  Rio de Janeiro  2020_1    0.0  0.0                 Morreram?  
1  Rio de Janeiro  2020_1    0.0  0.0  Olá novato, se apresente  
2  Rio de Janeiro  2020_1    0.0  0.0                 há tempos  
3  Rio de Janeiro  2020_1    0.0  0.0             Legião Urbana  
4  Rio de Janeiro  2020_1    0.0  0.0                    Indios  
-------------------------------------------info
Number of rows:  228618
Number of columns:  12


12 colunas, 228618 linhas

Colunas: <br>
id(Código de identificação) **INTEIRO**, <br>
date: (Data no formato xx/xx/xxxx) **DATA**, <br>
hora: (Horário no formato 24:00) **HORA**,<br>
ddi: (Discagem Direta Internacional) **NUMÉRICO**,<br>
country: (País) **STRING**,<br>
country_iso3: (Identificador de 3 letras do país) **STRING**,<br>
ddd: (Discagem Direta à Distância) **NUMÉRICO**,<br>
state: (Estado) **STRING**,<br>
group: (Ano e semestre) **STRING**,<br>
midia: (Bit indicador de Arquivos na mensagem) **BOOL**,<br>
url: (Bit indicador de links) **BOOL**,<br>
text: (conteúdo da mensagem) **STRING**

#### b) Identificar e listar as posições (células) contendo valores faltantes.
#### c)  Contar quantas linhas possuem valores faltantes.
#### d) Para cada coluna (feature), contar quantas linhas possuem valores faltantes.

In [6]:
import pandas as pd

data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')

mask = data1.isnull() #mask is a data frame with all the lines of the data, indicating which are null wiht a bool
missing_values = [(row,col) for row in data1.index for col in data1.columns if mask.at[row,col]]

print("\nNumber of missing values: ", len(missing_values))

print("\nNumber of missing values for each feature:\n", mask.sum(axis=0)) #calculates the sum of True isntances down the vertical axis for each column

print("\nPositions of missing values: ")
for i in missing_values:
    print("row: ", i[0], "|", "col: ", i[1] )



C:\Users\Renan\AppData\Local\Temp\ipykernel_1256\3235137087.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')



Number of missing values:  3872

Number of missing values for each feature:
 id                4
date            348
hour            349
ddi             349
country         351
country_iso3    351
ddd             351
state           351
group           349
midia           350
url             350
text            369
dtype: int64

Positions of missing values: 
row:  19 | col:  url
row:  19 | col:  text
row:  20 | col:  midia
row:  20 | col:  text
row:  21 | col:  group
row:  21 | col:  text
row:  22 | col:  state
row:  22 | col:  text
row:  23 | col:  ddd
row:  24 | col:  country_iso3
row:  25 | col:  country
row:  30 | col:  text
row:  32 | col:  text
row:  50 | col:  text
row:  51 | col:  state
row:  51 | col:  text
row:  52 | col:  ddd
row:  52 | col:  text
row:  53 | col:  country_iso3
row:  53 | col:  text
row:  55 | col:  country
row:  55 | col:  text
row:  67 | col:  text
row:  68 | col:  text
row:  69 | col:  text
row:  71 | col:  text
row:  72 | col:  text
row:  73 | col:  text

#### e) Identificar e listar as linhas repetidas (duplicadas)

In [7]:
import pandas as pd

data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')

duplicated_rows = data1.duplicated()

print("Number of duplicated rows=: ", duplicated_rows.sum())
print("\nDupliacted rows: \n", data1[duplicated_rows])

C:\Users\Renan\AppData\Local\Temp\ipykernel_1256\3937448060.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')


Number of duplicated rows=:  25614

Dupliacted rows: 
                           id      date   hour   ddi country country_iso3 ddd  \
68       8800670905318967146  07/04/20  18:25  55.0  BRASIL          BRA  14   
69       8800670905318967146  07/04/20  18:25  55.0  BRASIL          BRA  14   
72      -2588009083027728904  07/04/20  22:54  55.0  BRASIL          BRA  34   
73      -2588009083027728904  07/04/20  22:54  55.0  BRASIL          BRA  34   
92      -5563176891498971126  05/04/20  08:35  55.0  BRASIL          BRA  44   
...                      ...       ...    ...   ...     ...          ...  ..   
228568  -8734855083733576228  12/06/20  20:26  55.0  BRASIL          BRA  21   
228592  -8320603752794648767  12/06/20  21:48  55.0  BRASIL          BRA  63   
228603  -2262040611831006739  12/06/20  22:31  55.0  BRASIL          BRA  21   
228616  -2262040611831006739  12/06/20  22:53  55.0  BRASIL          BRA  21   
228617  -3678439454190211982  12/06/20  00:24  55.0  BRASIL      

#### f) Identificar e listar as posições (células) contendo valores que não pertencem ao domínio (tipo de dados) esperado.

In [45]:
import pandas as pd

data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')

#checking the current types for each feature
for col in data1.columns:
    col_type = data1[col].dtype
    print("Column '{}' has type: {}".format(col, col_type))

#Based on previous EDA, assigning expected types for each feature
expected_types = {
    'id': float, ##checked
    'date': 'datetime64[ns]', ##checked 
    'hour': object, ##checked
    'ddi': float, ##checked
    'country': str, ##kinda too hard to check besides seeing if its a number, assuming it's all checked up
    'country_iso3': str, ##checked
    ##'ddd': object, ##checked
    'state': str, ##kinda too hard to check besides seeing if its a number, assuming it's all checked up
    'group': str,  ##kinda too hard to check, assuming it's all checked up
    'midia': bool, ##checked
    'url': bool ##checked
    ##'text': objectnot really needed to check, can be anything a dataframe cacn hold
}

wrong_types_count = 0

#hour format 'xx:xx'
def is_valid_hour(hour_str):
    try:
        pd.to_datetime(hour_str, format='%H:%M')
        return True
    except ValueError:
        return False

#date format 'xx/xx/xx'
def is_valid_date(date_str):
    try:
        pd.to_datetime(date_str, format='%d/%m/%y')
        return True
    except ValueError:
        return False


for col, expected_type in expected_types.items():
    if col == 'hour':
        invalid_hours = [val for val in data1[col] if not is_valid_hour(val)]

        if invalid_hours:
            wrong_types_count += len(invalid_hours)
            for invalid_hour in invalid_hours:
                print(f"Invalid hour format in column '{col}': {invalid_hour}")

    elif col == 'date':
        invalid_dates = [val for val in data1[col] if not is_valid_date(val)]
        if invalid_dates:
            wrong_types_count += len(invalid_dates)
            for invalid_date in invalid_dates:
                print(f"Invalid date format in column '{col}': {invalid_date}")

    else:
        if col not in data1.columns:
            print(f"Column '{col}' not found.")
            continue

        for i, val in enumerate(data1[col]):
            #convertion attempt
            try:
                converted_val = expected_type(val)
            except (ValueError, TypeError):
                # If conversion fails, increment wrong types counter
                wrong_types_count += 1
                print(f"Wrong type at row {i}, column '{col}': {val}")

# Print total count of wroung types
print(f"Total wrong types: {wrong_types_count}")

C:\Users\Renan\AppData\Local\Temp\ipykernel_1256\159600256.py:3: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')


Column 'id' has type: object
Column 'date' has type: object
Column 'hour' has type: object
Column 'ddi' has type: float64
Column 'country' has type: object
Column 'country_iso3' has type: object
Column 'ddd' has type: object
Column 'state' has type: object
Column 'group' has type: object
Column 'midia' has type: float64
Column 'url' has type: float64
Column 'text' has type: object
Wrong type at row 177401, column 'id': *🌸🍃"❥❥ ̧ ̧.• ́ ̄❥❥•.°♡rick&morenah♡°.*
Wrong type at row 177402, column 'id': ▄▀▀▄▀▀▄░▄▀▀▄▀▀▄░▄▀▀▄▀▀▄
Wrong type at row 177403, column 'id': ▀▄░R░▄▀░▀▄░░░▄▀░▀▄░M░▄▀
Wrong type at row 177404, column 'id': ░░▀▄▀░░▒░░▀▄▀░░░░░▀▄▀░░
Wrong type at row 177405, column 'id': ██████████████████████
Wrong type at row 177406, column 'id': *•───── ୨❀୧ •─❤─୨❀୧ ─────•*
Wrong type at row 177407, column 'id': *•°.-ˏˋ ♡ ˊˎ-.°Seus olhos, sua voz, seu sorriso, seu cheiro, suas manias, seus abraços, seu jeito absurdo de me fazer tão sua! °.-ˏˋ ♡ ˊˎ-.°*
Wrong type at row 177408, column 'id': 

Colunas como texo basicamente podem receber qualquer obejto que o dataframe comporta, logo foi descartada da analise.
As colunas ddi e ddd, por meio de EDA e códigos anteriores foram bassicamente desccartadas da analise.
midia e url foram descartados da analise pro EDA.

Os únicos valores que não pertencem ao domínio da feature correspondente são os **344** valores encontrado, todos na coluna de identificação

#### g) Crie uma coluna chamada “caracteres” contendo a quantidade de caracteres de coluna “text”.

In [8]:
import pandas as pd

data1_1 = data1.assign(caracteres=data1['text'].str.len().fillna(0).astype(int))
data1_1.to_csv('data1_1.csv', index = True)

print(data1_1.head())

                    id      date   hour   ddi country country_iso3 ddd  \
0   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
1   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
2  5788106393468158140  07/04/20  04:07  55.0  BRASIL          BRA  21   
3   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
4  5788106393468158140  07/04/20  04:13  55.0  BRASIL          BRA  21   

            state   group  midia  url                      text  caracteres  
0  Rio de Janeiro  2020_1    0.0  0.0                 Morreram?           9  
1  Rio de Janeiro  2020_1    0.0  0.0  Olá novato, se apresente          24  
2  Rio de Janeiro  2020_1    0.0  0.0                 há tempos           9  
3  Rio de Janeiro  2020_1    0.0  0.0             Legião Urbana          13  
4  Rio de Janeiro  2020_1    0.0  0.0                    Indios           6  


#### h) Crie uma coluna chamada “words” contendo a quantidade de palavras da coluna “text”.

In [9]:
import pandas as pd

data1_2 = data1_1.assign(words=data1_1['text'].apply(lambda x: len(str(x).split()) if isinstance(x, str) else 0))
data1_2.to_csv('data1_2.csv', index = True)

print(data1_2.head())

print("-------------------------------------------info")

n_rows, n_col = data1_2.shape
print("Number of rows: ", n_rows)
print("Number of columns: ", n_col)

                    id      date   hour   ddi country country_iso3 ddd  \
0   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
1   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
2  5788106393468158140  07/04/20  04:07  55.0  BRASIL          BRA  21   
3   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
4  5788106393468158140  07/04/20  04:13  55.0  BRASIL          BRA  21   

            state   group  midia  url                      text  caracteres  \
0  Rio de Janeiro  2020_1    0.0  0.0                 Morreram?           9   
1  Rio de Janeiro  2020_1    0.0  0.0  Olá novato, se apresente          24   
2  Rio de Janeiro  2020_1    0.0  0.0                 há tempos           9   
3  Rio de Janeiro  2020_1    0.0  0.0             Legião Urbana          13   
4  Rio de Janeiro  2020_1    0.0  0.0                    Indios           6   

   words  
0      1  
1      4  
2      2  
3      2  
4      1  
--------------

#### i) Crie uma coluna chamada “viral” contendo o valor 0 se o texto da mensagem (valor do campo “text”) não for encontrado em outras linhas e 1, caso contrário.

In [11]:
import pandas as pd

data1_3 = data1_2.assign(viral=data1_2['text'].duplicated(keep=False)*1)
data1_3.to_csv('data1_3.csv', index = True)

print(data1_3.head())

print("-------------------------------------------info")

n_rows, n_col = data1_3.shape
print("Number of rows: ", n_rows)
print("Number of columns: ", n_col)

                    id      date   hour   ddi country country_iso3 ddd  \
0   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
1   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
2  5788106393468158140  07/04/20  04:07  55.0  BRASIL          BRA  21   
3   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
4  5788106393468158140  07/04/20  04:13  55.0  BRASIL          BRA  21   

            state   group  midia  url                      text  caracteres  \
0  Rio de Janeiro  2020_1    0.0  0.0                 Morreram?           9   
1  Rio de Janeiro  2020_1    0.0  0.0  Olá novato, se apresente          24   
2  Rio de Janeiro  2020_1    0.0  0.0                 há tempos           9   
3  Rio de Janeiro  2020_1    0.0  0.0             Legião Urbana          13   
4  Rio de Janeiro  2020_1    0.0  0.0                    Indios           6   

   words  viral  
0      1      1  
1      4      0  
2      2      0  
3      2

#### j) Crie uma coluna chamada “sharings” contendo a quantidade de vezes que o texto armazenado no atributo “text” aparece no dataset.

In [12]:
import pandas as pd

data1_4 = data1_3.assign(sharings=data1_3['text'].map(data1_3['text'].value_counts()).fillna(0).astype(int))
data1_4.to_csv('data1_4.csv', index = True)

print(data1_4.head())

print("-------------------------------------------info")

n_rows, n_col = data1_4.shape
print("Number of rows: ", n_rows)
print("Number of columns: ", n_col)

                    id      date   hour   ddi country country_iso3 ddd  \
0   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
1   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
2  5788106393468158140  07/04/20  04:07  55.0  BRASIL          BRA  21   
3   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
4  5788106393468158140  07/04/20  04:13  55.0  BRASIL          BRA  21   

            state   group  midia  url                      text  caracteres  \
0  Rio de Janeiro  2020_1    0.0  0.0                 Morreram?           9   
1  Rio de Janeiro  2020_1    0.0  0.0  Olá novato, se apresente          24   
2  Rio de Janeiro  2020_1    0.0  0.0                 há tempos           9   
3  Rio de Janeiro  2020_1    0.0  0.0             Legião Urbana          13   
4  Rio de Janeiro  2020_1    0.0  0.0                    Indios           6   

   words  viral  sharings  
0      1      1         2  
1      4      0         

#### k) Crie uma coluna chamada “sentiment” contendo os valores: -1 para textos negativos, 0 para textos neutros e 1 para textos positivos

In [28]:
from textblob import TextBlob
from googletrans import Translator
import pandas as pd

#gets the polarity atribute from a pt_text
def get_polarity(text_pt, n):

    global texts_count
    
    if not isinstance(text_pt, str):
        return None
    
    try:

        if texts_count >= n:
            return None
        texts_count += 1

        if len(str(text_pt)) > 430:
            ##print(f"Skipping text with length {len(text_pt)} characters: {text_pt}")
            return -1
        
        #translates the text so it can be used on textblob
        translator = Translator()
        text_en = translator.translate(text_pt, src='pt', dest='en').text

        polarity = TextBlob(text_en).sentiment.polarity

        if polarity <= -0.5:
            return -1
        elif -0.5 < polarity < 0.5:
            return 0
        else:
            return 1
    
    except Exception as e:
        print(f"Error occurred during translation: {e}")
        return None


texts_count = 0
n = 100

data1_5 = data1_4.head(n).assign(sentiment=data1_4['text'].head(n).apply(get_polarity, n=n))
data1_5.to_csv('data1_5.csv', index = True)
print(data1_5.head())


Error occurred during translation: list index out of range
                    id      date   hour   ddi country country_iso3 ddd  \
0   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
1   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
2  5788106393468158140  07/04/20  04:07  55.0  BRASIL          BRA  21   
3   146759200457638065  07/04/20  04:07  55.0  BRASIL          BRA  21   
4  5788106393468158140  07/04/20  04:13  55.0  BRASIL          BRA  21   

            state   group  midia  url                      text  caracteres  \
0  Rio de Janeiro  2020_1    0.0  0.0                 Morreram?           9   
1  Rio de Janeiro  2020_1    0.0  0.0  Olá novato, se apresente          24   
2  Rio de Janeiro  2020_1    0.0  0.0                 há tempos           9   
3  Rio de Janeiro  2020_1    0.0  0.0             Legião Urbana          13   
4  Rio de Janeiro  2020_1    0.0  0.0                    Indios           6   

   words  viral  shar

#### l) Eliminar as linhas cujo valor da coluna “text” contenham “trava-zaps”.

In [31]:
def remove_trava_zaps(data):
    import re
    trava_zaps_i = []
    for i, row in data.iterrows():
        if 'wa.me' in str(row['text']): 
            trava_zaps_i.append(i)
            print(f"Trava zap found at index {i}: {row}")
    print("Number of trava_zaps found: ", len(trava_zaps_i))
    
    return data.drop(index=trava_zaps_i)


data1_6 = remove_trava_zaps(data1_4)
data1_6.to_csv('data1_6.csv', index = True)
print(data1_6.head())

Trava zap found at index 1092: id                                           -6575413577977330488
date                                                     22/03/20
hour                                                        05:57
ddi                                                          55.0
country                                                    BRASIL
country_iso3                                                  BRA
ddd                                                            21
state                                              Rio de Janeiro
group                                                      2020_5
midia                                                         0.0
url                                                           1.0
text            _FORMANDO CONDUTORES EM TODO BRASIL MAIS DETAL...
caracteres                                                   1658
words                                                         244
viral                                        

#### m) Identificar inconsistências entre os atributos (features).

In [49]:
import pandas as pd

print(data1_6.dtypes)
print("-------------------------------------------------------------------")
print(data1_6.describe())
print("-------------------------------------------------------------------")

data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')

# Checking the current types of each feature
for col in data1.columns:
    col_type = data1[col].dtype
    print("Column '{}' has type: {}".format(col, col_type))

# Based on previous EDA, assigning expected types for each feature
expected_types = {
    'id': float, 
    'date': 'datetime64[ns]', 
    'hour': object, 
    'ddi': float, 
    'country': str,
    'country_iso3': str,
    'ddd': str,  # Changing the expected type of 'ddd' to string
    'state': str,
    'group': str,
    'midia': bool,
    'url': bool
}

# Initialize counter for wrong types
wrong_types_count = 0

# Function to check if a string is a valid date
def is_valid_date(date_str):
    try:
        pd.to_datetime(date_str, format='%d/%m/%y')  # Changed the format string to '%d/%m/%y'
        return True
    except ValueError:
        return False
    
def is_valid_hour(hour_str):
    try:
        pd.to_datetime(hour_str, format='%H:%M')
        return True
    except ValueError:
        return False

def is_valid_country_iso3(iso3_str):
    return len(iso3_str) == 3 if pd.notna(iso3_str) and isinstance(iso3_str, str) else True

def is_valid_ddd(ddd_str):
    return len(ddd_str) <= 2 if pd.notna(ddd_str) and isinstance(ddd_str, str) else True

# Iterate over each column
for col, expected_type in expected_types.items():
    # Special handling for hour column
    if col == 'hour':
        # Check if all values in the hour column are in the format 'xx:xx'
        invalid_hours = [val for val in data1[col] if not is_valid_hour(val)]
        if invalid_hours:
            wrong_types_count += len(invalid_hours)
            for invalid_hour in invalid_hours:
                print(f"Inconsistency at '{col}': {invalid_hour}")
    elif col == 'date':
        # Check if all values in the date column are in the correct format
        invalid_dates = [val for val in data1[col] if not is_valid_date(val)]
        if invalid_dates:
            wrong_types_count += len(invalid_dates)
            for invalid_date in invalid_dates:
                print(f"Inconsistency at '{col}': {invalid_date}")
    elif col == 'country_iso3':
        # Check if all values in the country_iso3 column have exactly three letters
        invalid_iso3 = [iso3 for iso3 in data1[col] if not is_valid_country_iso3(iso3)]
        if invalid_iso3:
            wrong_types_count += len(invalid_iso3)
            for invalid_value in invalid_iso3:
                print(f"Inconsistency at '{col}': {invalid_value}")
    elif col == 'ddd':
        # Check if all values in the ddd column have exactly two digits
        invalid_ddd = [ddd for ddd in data1[col] if not is_valid_ddd(ddd)]
        if invalid_ddd:
            wrong_types_count += len(invalid_ddd)
            for invalid_value in invalid_ddd:
                print(f"Inconsistency at '{col}': {invalid_value}")
    else:
        # Check if the column exists
        if col not in data1.columns:
            print(f"Column '{col}' not found.")
            continue

        # Iterate over each cell in the column
        for i, val in enumerate(data1[col]):
            # Attempt to convert the value to the expected type
            try:
                converted_val = expected_type(val)
            except (ValueError, TypeError):
                # If conversion fails, increment wrong types counter
                wrong_types_count += 1
                print(f"Inconsistency at {i}, column '{col}': {val}")

# Print total count of wrong types
print(f"Inconsistencies: {wrong_types_count}")



id               object
date             object
hour             object
ddi             float64
country          object
country_iso3     object
ddd              object
state            object
group            object
midia           float64
url             float64
text             object
caracteres        int32
words             int64
viral             int32
sharings          int32
dtype: object
-------------------------------------------------------------------
                 ddi          midia            url     caracteres  \
count  228066.000000  228065.000000  228065.000000  228415.000000   
mean       57.047653       0.328990       0.093175     151.813384   
std        28.961813       0.469847       0.290678    1952.373642   
min         1.000000       0.000000       0.000000       0.000000   
25%        55.000000       0.000000       0.000000      12.000000   
50%        55.000000       0.000000       0.000000      25.000000   
75%        55.000000       1.000000       0.000000 

C:\Users\Renan\AppData\Local\Temp\ipykernel_1256\2182802273.py:8: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  data1 = pd.read_csv('covid19br_vt1.csv', delimiter=';')


Column 'id' has type: object
Column 'date' has type: object
Column 'hour' has type: object
Column 'ddi' has type: float64
Column 'country' has type: object
Column 'country_iso3' has type: object
Column 'ddd' has type: object
Column 'state' has type: object
Column 'group' has type: object
Column 'midia' has type: float64
Column 'url' has type: float64
Column 'text' has type: object
Wrong type at row 177401, column 'id': *🌸🍃"❥❥ ̧ ̧.• ́ ̄❥❥•.°♡rick&morenah♡°.*
Wrong type at row 177402, column 'id': ▄▀▀▄▀▀▄░▄▀▀▄▀▀▄░▄▀▀▄▀▀▄
Wrong type at row 177403, column 'id': ▀▄░R░▄▀░▀▄░░░▄▀░▀▄░M░▄▀
Wrong type at row 177404, column 'id': ░░▀▄▀░░▒░░▀▄▀░░░░░▀▄▀░░
Wrong type at row 177405, column 'id': ██████████████████████
Wrong type at row 177406, column 'id': *•───── ୨❀୧ •─❤─୨❀୧ ─────•*
Wrong type at row 177407, column 'id': *•°.-ˏˋ ♡ ˊˎ-.°Seus olhos, sua voz, seu sorriso, seu cheiro, suas manias, seus abraços, seu jeito absurdo de me fazer tão sua! °.-ˏˋ ♡ ˊˎ-.°*
Wrong type at row 177408, column 'id': 